packages/jsonapi/middleware.js

```diff
@@ -218,15 +218,24 @@ class Handler {
 
   async handleIndividualPATCH(type, id) {
     let data = this._mandatoryBodyData();
-    let record = await this.writers.update(this.session, type, id, data);
+    let record;
+    if (type === 'cards') {
+      record = await this.cardServices.update(this.session, id, data);
```

Comment:
Why is this needed? I was under the impression you had already solved the same problem inside hub/writers instead. We should solve it at this layer or that layer but probably not both?

Reply:
i did, but I thought for consistency that within the middleware we always go through the CardServices layer to interact with cards. (since we do that for the get-invididual and get-collection methods), ditto for delete.

Reply:
Card service's mutation methods just wraps the writer functions for create, update, and delete and assumes a `cards` type

Reply:
If you'd prefer not to do this, then no biggie, I can remove these wrapped methods.

Reply:
Oh yeah, I see the consistency concern.

I think if we're going to create the new interface at the level of hub/writers, then we also need to do it in hub/searchers. Those two are paired.

The alternative is to do it in jsonapi middleware and then *not* in hub/writers and hub/searchers. But I think that's probably less good, since there are plenty of other use cases (like tests) that want to go directly into hub/writers and hub/searchers.